go/vt/discovery/healthcheck.go

```diff
@@ -741,11 +742,12 @@ func (hc *HealthCheckImpl) WaitForInitialStatsUpdates() {
 // GetConnection returns the TabletConn of the given tablet.
 func (hc *HealthCheckImpl) GetConnection(key string) queryservice.QueryService {
 	hc.mu.RLock()
-	defer hc.mu.RUnlock()
 	hcc := hc.addrToConns[key]
 	if hcc == nil {
+		hc.mu.RUnlock()
 		return nil
 	}
+	hc.mu.RUnlock()
```

Comment:
This fix can cause problems because hcc.Conn can become nil after hc.mu.RUnlock, and Getconnection will end up returning nil.

Obtaining the read lock before releasing hc.mu will work. But then, we might as well leave it like before.

Reply:
As discussed on Slack I don't actually think this changes the semantics of whether or not hcc.Conn is nil.